fw/apm.c

```diff
@@ -1375,9 +1384,15 @@ tfw_apm_hm_srv_alive(int status, TfwStr *body, struct sk_buff *skb_head,
 	 * from body of first response and store it into monitor.
 	 */
 	if (!hm->crc32 && hm->auto_crc) {
-		hm->crc32 = __tfw_apm_crc32_calc(&it, &chunk, skb_head, body);
+		hm->crc32 = __tfw_apm_crc32_calc(&it, &chunk);
+		/*This warinig is used by tests to check crc generation
+		 * '\n' is neseccary for flushing message to logger,
+		 * without it tests will always fail.  
+		 */
+		T_WARN("Auto CRC generated\n");
```

Comment:
Is it necessary to have this WARN only for testing? Can we verify correctness of the code relying on state of the backend? For example: If backend was dead and then it became alive, it means all works. Then do negative tests with wrong crc where expected result is dead server.

Reply:
May be it is not the best way to test it, but overhead looks as reasonable. We get the single warning message at the moment when HM start working. The message explicitly indicate CRC generation.
In case of initially dead backend we:
1. Add delay before test (@RomanBelozerov  asked me to minimize test execution time, as it possible)
2. Change single warning to backend on/off  warnings. 
3. Indicate proper works of subsystem at all, but not CRC issues. Using framework we can check only backend CRC, which is not our responsibility.

Reply:
@const-t
I though a lot about it and maybe make sense to introduce separate logging macros which will be active only for CI builds and print out information only for test framework?

Reply:
>  I though a lot about it and maybe make sense to introduce separate logging macros which will be active only for CI builds and print out information only for test framework?

I would like to suggest to use [dynamic debug](https://www.kernel.org/doc/html/v4.19/admin-guide/dynamic-debug-howto.html) for such debug prints. I've used it and it works perfect. @krizhanovsky  @RomanBelozerov Guys, what do you think?

Reply:
I'm strongly against the warning used for testing only. The most bad thing is that this is *warning* for absolutely normal cast to compute CRC, even not an informational message, - it will flood the user logs and the user won't be able to figure out what's wrong. This is absolute not go.

Probably, it's it's to hard to test in automatic way, we can just test it manually. But I don't see why it's impossible to implement the test in Python. @RomanBelozerov seems some help from your side is required here